express-zod-api/src/deep-checks.ts

```diff
@@ -34,6 +34,30 @@ export const findNestedSchema = (
     (err: DeepCheckError) => err.cause,
   )();
 
+/** not using cycle:"throw" because it also affects parenting objects */
+export const hasCycle = (
+  subject: $ZodType,
+  { io }: Pick<NestedSchemaLookupProps, "io">,
+) => {
+  const json = z.toJSONSchema(subject, {
+    io,
+    unrepresentable: "any",
+    override: ({ jsonSchema }) => {
+      if (typeof jsonSchema.default === "bigint") delete jsonSchema.default;
+    },
+  });
+  const stack: unknown[] = [json];
+  while (stack.length) {
+    const entry = stack.shift()!;
+    if (R.is(Object, entry)) {
+      if ((entry as JSONSchema.BaseSchema).$ref === "#") return true;
+      stack.push(...R.values(entry));
+    }
+    if (R.is(Array, entry)) stack.push(...R.values(entry));
+  }
```

Comment:
DNRY with `fixReferences`, consider extracting a JSON Schema helper for such traverse

Reply:
does not worth it, too different